openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,212 @@
+    previously and denotes that representation files should be converted.
+    This dict contains source colorspace information, collected by hosts.
+
+    Target colorspace is selected by profiles in the Settings, based on:
+    - families
+    - host
+    - task types
+    - task names
+    - subset names
+    """
+
+    label = "Transcode color spaces"
+    order = pyblish.api.ExtractorOrder + 0.019
+
+    optional = True
+
+    # Supported extensions
+    supported_exts = ["exr", "jpg", "jpeg", "png", "dpx"]
+
+    # Configurable by Settings
+    profiles = None
+    options = None
+
+    def process(self, instance):
+        if not self.profiles:
+            self.log.debug("No profiles present for color transcode")
+            return
+
+        if "representations" not in instance.data:
+            self.log.debug("No representations, skipping.")
+            return
+
+        if not is_oiio_supported():
+            self.log.warning("OIIO not supported, no transcoding possible.")
+            return
+
+        profile = self._get_profile(instance)
+        if not profile:
+            return
+
+        repres = instance.data.get("representations") or []
+        for idx, repre in enumerate(list(repres)):
+            self.log.debug("repre ({}): `{}`".format(idx + 1, repre["name"]))
+            if not self._repre_is_valid(repre):
+                continue
+
+            colorspace_data = repre["colorspaceData"]
+            source_color_space = colorspace_data["colorspace"]
+            config_path = colorspace_data.get("configData", {}).get("path")
+            if not os.path.exists(config_path):
+                self.log.warning("Config file doesn't exist, skipping")
+                continue
+
+            repre = self._handle_original_repre(repre, profile)
+
+            for _, output_def in profile.get("outputs", {}).items():
+                new_repre = copy.deepcopy(repre)
+
+                new_staging_dir = get_transcode_temp_directory()
+                original_staging_dir = new_repre["stagingDir"]
+                new_repre["stagingDir"] = new_staging_dir
+                files_to_convert = new_repre["files"]
+                if not isinstance(files_to_convert, list):
+                    files_to_convert = [files_to_convert]
+
+                files_to_delete = copy.deepcopy(files_to_convert)
+
+                output_extension = output_def["output_extension"]
+                output_extension = output_extension.replace('.', '')
+                if output_extension:
+                    new_repre["name"] = output_extension
+                    new_repre["ext"] = output_extension
+
+                    files_to_convert = self._rename_output_files(
+                        files_to_convert, output_extension)
+
+                files_to_convert = [os.path.join(original_staging_dir, path)
+                                    for path in files_to_convert]
```

Comment:
Logic here is wrong. The `files_to_convert` would've already been remapped to a new extension above but the files on disk have not. So we'd from here on out be looking in the original staging dir for files which not exist.